### PR TITLE
MRG: compatibility with bagging classifiers

### PR DESCRIPTION
**Pull request by kingjr**
Fix bug when using ensemble estimators with `SlidingEstimator` or `GeneralizingEstimator` caused by sklearn internal inconsistency, https://github.com/scikit-learn/scikit-learn/issues/9720

### CODE REVIEW

File: mne/decoding/search_light.py [hunk 1]
```diff
@@ -82,7 +82,12 @@ def fit(self, X, y, **fit_params):
         estimators = parallel(
             p_func(self.base_estimator, split, y, **fit_params)
             for split in np.array_split(X, n_jobs, axis=-1))
-        self.estimators_ = np.concatenate(estimators, 0)
+        self.estimators_ = np.empty(X.shape[-1], dtype=object)
```
Comment (agramfort) [inline mne/decoding/search_light.py]: can you add a note why this for loops are needed?

we could not make it work with lists? ie why self.estimators_ needs to be an array?

Comment (kingjr) [reply mne/decoding/search_light.py]: You want me to add an explicit comment on sklearn's bad design its devs unwillingness to correct for it? ;)

> we could not make it work with lists? i.e why self.estimators_ needs to be an array?

mmh, do you mean, to reduce the number of lines? I'm -1 to change the data type for such a small gain if we don't see a clear advantage on the user side.

Ultimately, this `SearchLight` should become > 1 dimensional, to meet the other packages' functionality (i.e. search light in 2D (eg. time frequency) or 3D (anatomical sources)). There, having an array greatly facilitate the data manipulation.

Comment (kingjr) [reply mne/decoding/search_light.py]: (BTW, the for loop costs is null, it's just to re-shape the array-split induced by joblib)

**Comment by kingjr** [comment]
In sum
* making `estimators_` a list instead of an array doesn't solve the BaggingClassifier issue, which is induced because sklearn `BaseEnsemble` is an iterable that give its internal `estimators_` which creates a conflict with joblib arraysplit
* The simpler concatenation via e.g. `np.concatenate` and `[est for job in estimators for est in job]` don't work.

**Comment by agramfort** [comment]
thx @kingjr
